vendor/k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go

```diff
@@ -49,6 +49,11 @@ func ValidateLabelSelectorRequirement(sr metav1.LabelSelectorRequirement, fldPat
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("operator"), sr.Operator, "not a valid selector operator"))
 	}
 	allErrs = append(allErrs, ValidateLabelName(sr.Key, fldPath.Child("key"))...)
+	for _, v := range sr.Values {
```

Comment:
I don't think this is the right approach, since this might introduce inconsistencies with newer versions.

Reply:
Looking at this further, I think this is the error and it needs to be ported to all available versions (see https://github.com/kubernetes/kubernetes/pull/92714). There is currently no guaranteed handling of an invalid selector and this should be caught at creation time